### PR TITLE
[MBL-1607] Add Estimated Shipping View to Late Pledge Checkout

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Adds a new SwiftUI to display estimated shipping cost

# 🤔 Why

[JIRA TICKET](https://kickstarter.atlassian.net/browse/MBL-1607)

# 🛠 How

- Adds the new SwiftUI View `EstimatedShippingCheckoutView` to `PostCampaignCheckoutViewController`
- Add snapshot tests

# 👀 See

thisone

# ✅ Acceptance criteria

- [x] New view matches [the new designs](https://www.figma.com/design/z4faZM4BQwbyiDe7y8tbpP/Pledge-Redemption?node-id=105-8312&t=x8MhTSiTedcRIb7a-4)
- [x] If there is no estimated shipping or if shipping is not enabled for the selected reward, we don't show this new view.

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/ViewModels/ConfirmDetailsViewModel.swift:348:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/ViewModels/NoShippingConfirmDetailsViewModel.swift:289:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by ifosli — APPROVED** [review]
I'm concerned about how this is making the late pledge VC layout worse, even when there's no estimated shipping. Can you confirm that Alison is okay with launching with this layout (extra left and right margins for the pledge view and weird gap between the scroll view and the pledge cta) even for late pledges without estimated shipping? If not, I would strongly recommend fixing it in this pr but I'm also okay with filing a follow-up ticket/fixing it in a separate pr (follow-up ticket should probably happen anyways so we can fix it for the next release, if it ends up not being launch blocking).

File: Kickstarter-iOS/Features/PledgeView/Controllers/NoShippingPledgeViewController.swift [hunk 1]
```diff
@@ -108,6 +108,11 @@ final class NoShippingPledgeViewController: UIViewController,
       |> \.translatesAutoresizingMaskIntoConstraints .~ false
   }()
 
+  private lazy var estimatedShippingStackView: UIStackView = {
+    UIStackView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
```
Comment (scottkicks) [inline Kickstarter-iOS/Features/PledgeView/Controllers/NoShippingPledgeViewController.swift:114]: @ifosli For the correct margins to be applied, the estimated shipping view needs to be added to a new stack view as a subview. This is true for UIKit views as well. I spent time rebuilding the view with UIKit to verify this.

**Pull request merged**